internal/pkg/aws/ecs/ecs.go

```diff
@@ -99,6 +109,60 @@ func (e *ECS) Service(clusterName, serviceName string) (*Service, error) {
 	return nil, fmt.Errorf("cannot find service %s", serviceName)
 }
 
+// UpdateServiceOpts sets the optional parameter for UpdateService.
+type UpdateServiceOpts func(*ecs.UpdateServiceInput)
+
+// WithForceUpdate sets ForceNewDeployment to force an update.
+func WithForceUpdate() UpdateServiceOpts {
+	return func(in *ecs.UpdateServiceInput) {
+		in.ForceNewDeployment = aws.Bool(true)
+	}
+}
+
+// UpdateService calls ECS API and updates the specific service running in the cluster.
+func (e *ECS) UpdateService(clusterName, serviceName string, opts ...UpdateServiceOpts) error {
+	in := &ecs.UpdateServiceInput{
+		Cluster: aws.String(clusterName),
+		Service: aws.String(serviceName),
+	}
+	for _, opt := range opts {
+		opt(in)
+	}
+	svc, err := e.client.UpdateService(in)
+	if err != nil {
+		return fmt.Errorf("update service %s from cluster %s: %w", serviceName, clusterName, err)
+	}
+	s := Service(*svc.Service)
+	if err := e.waitUntilServiceStable(&s); err != nil {
+		return fmt.Errorf("wait until service %s becomes stable: %w", serviceName, err)
+	}
+	return nil
+}
+
+// waitUntilServiceStable waits until the service is stable.
+// See https://docs.aws.amazon.com/cli/latest/reference/ecs/wait/services-stable.html
+func (e *ECS) waitUntilServiceStable(svc *Service) error {
+	var err error
+	defer func() {
+		e.tryNum = 0
+	}()
+	for {
+		if len(svc.Deployments) == stableServiceDeploymentNum &&
```

Comment:
Is it possible that immediately after calling `UpdateService` there is eventual consistency issue and the number of deployments is shown as `1` while there are `2` that's supposed to happen?

The way we get around it in the progress tracker is by comparing the deployment's timestamp with the timestamp of the request. (it has to be after)
https://github.com/aws/copilot-cli/blob/3c0b1d4bc8ba6b2beb153891752fd62ae505d93b/internal/pkg/stream/ecs.go#L230-L232

Reply:
> Is it possible that immediately after calling UpdateService there is eventual consistency issue and the number of deployments is shown as 1 while there are 2 that's supposed to happen?

That sounds like an ECS bug to me. I haven't had this issue. Did you have this issue before? But sure I agree we should be safer on our end.

Reply:
I'm struggling to remember honestly. I ran into this while working on the progress tracker but that might have been because we're updating via cloudformation instead of directly with the SDK.